addons/project/static/src/views/project_task_kanban/project_task_kanban_compiler.js

```diff
@@ -0,0 +1,50 @@
+/** @odoo-module **/
+
+import { KanbanCompiler } from "@web/views/kanban/kanban_compiler";
+import { append, createElement } from "@web/core/utils/xml";
+
+export class ProjectTaskKanbanCompiler extends KanbanCompiler {
+    setup() {
+        super.setup();
+        this.compilers.push(
+            { selector: ".subtask_list_button", fn: this.compileSListButton },
+            { selector: "div.kanban_bottom_subtasks_section", fn: this.compileSListComponent },
+        );
+    }
+
+    /**
+     * @param {Element} el
+     * @returns {Element}
+     */
+    compileSListButton(el) {
+        el.setAttribute("t-on-click", `() => state.folded = !state.folded`);
+        el.removeAttribute("class");
+        el.setAttribute("t-attf-class", "subtask_list_button btn fa {{ state.folded ? 'fa-solid fa-angle-right' : 'fa-solid fa-angle-down' }}");
+        const compiled = createElement(el.nodeName);
+        for (const { name, value } of el.attributes) {
+            compiled.setAttribute(name, value);
+        }
+
+        return compiled;
+    }
+
+    /**
+     * @param {Element} el
+     * @returns {Element}
+     */
+    compileSListComponent(el) {
+        el.setAttribute("t-if", `!state.folded and !selection_mode`);
+        const compiled = createElement(el.nodeName);
+        for (const { name, value } of el.attributes) {
+            compiled.setAttribute(name, value);
+        }
+        const listContainer = createElement('widget');
+        const listElemenent = createElement('SubtaskKanbanList');
+        listElemenent.setAttribute("record", '__comp__.props.record');
+
+        append(listContainer, listElemenent);
+        append(compiled, listContainer);
+
+        return compiled;
+    }
+}
```

Comment:
To check the subtaskList component are compiled with the button, we will have the override the main method called to compile the view.
This main method is `compile` one.

```suggestion
    compile(key, params = {}) {
        const newRoot = super.compile(key, params);
        if (this.subtaskListComponentCompiled.component !== this.subtaskListComponentCompiled.button) {
            // Error since one of them is not compiled
            throw new Error("The subtask list component cannot be rendered if the button and the component are not in the view definition.");
        }
        return newRoot;
    }
}
```

(I don't think we have to translate that since it is a Dev error) 🤔